packages/grid/theme/material/vaadin-grid-styles.js

```diff
@@ -55,15 +55,18 @@ registerStyles(
     /* Body rows/cells */
 
     [part~='body-cell'] {
-      background-color: var(--material-background-color);
+      background-color: var(--vaadin-grid-cell-background, var(--material-background-color));
     }
 
     [part~='row']:hover > [part~='body-cell'] {
-      background: linear-gradient(
-          var(--_material-grid-row-hover-background-color, rgba(0, 0, 0, 0.04)),
-          var(--_material-grid-row-hover-background-color, rgba(0, 0, 0, 0.04))
-        )
-        var(--material-background-color);
+      background: var(
+        --vaadin-grid-cell-background,
+        linear-gradient(
+            var(--_material-grid-row-hover-background-color, rgba(0, 0, 0, 0.04)),
+            var(--_material-grid-row-hover-background-color, rgba(0, 0, 0, 0.04))
+          )
+          var(--material-background-color)
+      );
```

Comment:
I added a comment on the original ticket about this: https://github.com/vaadin/web-components/issues/6491#issuecomment-1787089774